testCurrentRangeDetection.py

```diff
@@ -0,0 +1,30 @@
+import unittest
+import currentRangeDetection
+
+class currentRangeDetectionTest(unittest.TestCase):
+  def test_IsRangeOk(self):
+    self.assertTrue(currentRangeDetection.IsRangeOk([0,99]) == True)
+    self.assertFalse(currentRangeDetection.IsRangeOk([-1,99]) == False)
+    self.assertFalse(currentRangeDetection.IsRangeOk([0,100]) == False)
+
+  def test_get_sorted_Samples(self):
+    self.assertTrue(currentRangeDetection.get_sorted_Samples([3,5,4]) == [3,4,5])
+    self.assertTrue(currentRangeDetection.get_sorted_Samples([5,6,6,7,1]) == [1,5,6,6,7])
+    self.assertFalse(currentRangeDetection.get_sorted_Samples([5,6,6,7,1]) == [1,5,6,7])
+
+  def test_get_CurrentRangeslist(self):

```

Comment:
Excellent way to express the stages in the algorithm via tests 👍 
Only suggestion is to name the test function by its functionality.
E.g., instead of `test_get_CurrentRangeslist`, name it as `test_sorted_samples_to_continuous_ranges`

Reply:
Thanks for the suggestion.
Renamed the test functions as below
- test_samples_IsRangeOk
- test_sorted_Samples
- test_sorted_samples_to_continuous_ranges
-  test_csv_output
- test_currentvalues_to_csv_output
- test_functionality

Reply:
@baburoshima good rework and responses 👍 
See if you can include the input and output in all the test names - like you've done in the tests with `_to_` in them.